### PR TITLE
Minor HttpClient improvement. Mutable bufferSize.

### PR DESCRIPTION
**Pull request by asurovenko**
Hi!
Please tell us more about `BUFFER_SIZE=8000` meaning. What if response bigger than 8000 bytes? If I'm sure that **almost** all my responses have less than 6000 bytes, does it mean that I should set `BUFFER_SIZE=6000`?

### CODE REVIEW

**Comment by apangin** [comment]
It's just a buffer size, like in `BufferedInputStream`. No matter how large a response is, `HttpClient` should work correctly with a smaller or a larger buffer. The only requirement is that the buffer is large enough to hold at least one line of response header.

It's OK to make the buffer size configurable. You can even add a corresponding `ConnectionString` parameter (that's exactly what ConnectionString was designed for).

**Comment by asurovenko** [comment]
I don't understand the point about `ConnectionString`. You mean make the setter for `ConnectionString`? For what? I can't imagine case in that we must reuse client.

**Comment by apangin** [comment]
`ConnectionString` is used to set up SocketPool and HttpClient parameters. Since you are adding a tunable parameter to HttpClient, it might be a good idea to configure it via ConnectionString as well.

File: src/one/nio/http/HttpClient.java [hunk 1]
```diff
@@ -234,6 +239,20 @@ Response readResponse(int method) throws IOException, HttpException {
...
+
+            do {
+                if (pos >= length) {
+                    throw new HttpException("Response header absent");
+                }
+            } while (buf[pos++] != '\n');
+
+            this.pos = pos;
+            return Utf8.read(buf, 9, pos - 2);
+        }
```
Comment (apangin) [inline src/one/nio/http/HttpClient.java]: I'm not sure why this function has been added, but it definitely looks incorrect. There is no guarantee that the first line of the response will be received entirely in the first packet.

Comment (asurovenko) [reply src/one/nio/http/HttpClient.java]: But how I can see in code: first line always get as is and then it's cut on `.substring(9)`
Why we need extra String? If you always cut it, then, why you need to create it with the first 9 bytes?

Comment (apangin) [reply src/one/nio/http/HttpClient.java]: This is just for simplicity and for code reuse. If we see this part a performance bottleneck, we'll optimize it.

**Comment by apangin** [comment]
Closing PR due to inactivity.
`bufferSize` is made configurable via `ConnectionString` parameter.

**Pull request closed without merging**